### PR TITLE
Natural language processing using TensorFlow and Bert Model

### LINKED ISSUE #7179: [Machine Learning] Natural language processing using TensorFlow and Bert Model
#### Proposed title of article
[Machine Learning] Natural language processing using TensorFlow and Bert Model

#### Proposed article introduction
Natural language processing is the application of computational techniques to the analysis and synthesis of natural language and speech. Hugging Face provides thousands of pre-trained models to perform tasks on different modalities such as text, vision, and audio. Text, for tasks like text classification, information extraction, question answering, summarization, translation, and text generation. Images, for tasks like image classification, object detection, and segmentation. Audio, for tasks like speech recognition and audio classification.

Hugging Face Transformers support pre-trained models such as ALBERT, BERT, BART, Barthez, etc. In this tutorial, we will be focusing on the BERT model. Bidirectional Encoder Representations from Transformers (BERT) is a transformer-based machine learning technique for natural language processing (NLP) pre-training developed by Google. BERT is designed to help computers understand the meaning of ambiguous language in the text by using surrounding text to establish context.

We import BERT from the hugging face transformers. We will use the model to build a custom sentiment analysis. Sentiment analysis is an NLP task that classifies customers reviews. The reviews can be: `0 - negative`, `1 - somewhat negative`,  `2 - neutral`,  `3 - somewhat positive` and `4 - positive`. We will use TensorFlow to fine-tune the model.

#### Key takeaways 
1. Dataset preparation.
2. Installing the HuggingFace transformers.
3. EXploring BERT functions.
4. Importing BertTokenizer
5. Building the sentiment analysis model using TFBertModel.
6. Making predictions.

#### Article quality
This article is unique because it explains the concept of transformers in detail. Using the article the reader will be able to know the different transformer models before focusing on BERT. It also covers the basic BERT architecture, this will enable the reader to know how to fine-tune the model to understand sentiment analysis. The article will also cover HuggingFaceTransforemers, we will discuss the different tasks that the libraries can solve. The tutorial uses detailed steps that a reader can easily follow.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by charlesndirutu33**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #7179

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by wilkistermumbi** [comment]
@charlesndirutu33 It is always good practice to perform a Grammarly check on your article once you're done with the technical writing. Make sure you use the premium version of Grammarly.

**Comment by charlesndirutu33** [comment]
@wilkistermumbi, I have performed a Grammarly check on my article. Kindly review thanks.

File: content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+### What is Hugging Face Transformers?
+Hugging Face Transformers provides APIs to download and fine-tune pre-trained models. 
+
+There are various pre-trained models for NLP tasks, image classification, video classification, and audio classification. It supports different pre-trained models such as BERT. 
+
+To see all the supported pre-trained models, click [here](https://huggingface.co/docs/transformers/index)
+
+Hugging Face Transformers can easily integrate with machine learning libraries such as [Pytorch](https://pytorch.org/) and [TensorFlow](tensorflow.org).
+
+We will start by installing the Hugging Face Transformers.
```
Comment (wilkistermumbi) [inline content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md:67]: As a rule of thumb, make sure your sentences are at least three lines long. Fix this issue throughout your article.

File: content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+
+#### Installing Hugging Face Transformers
+We use the following code:
```
Comment (wilkistermumbi) [inline content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md:70]: To install.......we use......

Please fix such sentences.

File: content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+**tqdm**
+We use this library to create input tensors that the model will use.
+
+**tensorflow**
+We will use TensorFlow to train the model. It will also add all the layers the model requires.
+
+After the installation, let's start working with the dataset for sentiment analysis.
+
+### Working with sentiment analysis dataset
+We will use a movie revies dataset that has different sentiment labels.
```
Comment (wilkistermumbi) [inline content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md:94]: Always proofread your work. Please fix this sentence.

File: content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md [hunk 4]
```diff
@@ -0,0 +1,560 @@
...
+```python
+import pandas as pd
+```
+To read the sentiment analysis dataset, use this code:
+
+```python
+df = pd.read_csv('/content/train.tsv', sep='\t')
+```
+The dataset has tab-separated values (TSV). 
+Let's display the dataset in the Google Colab notebook.
```
Comment (wilkistermumbi) [inline content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md:126]:
```suggestion
The dataset has tab-separated values (TSV). Let's display the dataset in the Google Colab notebook.
```

File: content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+
+```pyton
+model = TFBertModel.from_pretrained('bert-base-cased')
+```
+The code above initializes the `TFBertModel`. It also downloads the `bert-base-cased` model that will perform sentiment analysis. 
+
+The next step is to add the input, intermediate, and output layers to the `TFBertModel` model.
+
+### Adding layers
+We will use TenerFlow's Keras to add all the required layers for the model. Let's first add the input layers.
```
Comment (wilkistermumbi) [inline content/articles/natural-language-processing-using-tensorflow-and-bert-model/index.md:347]: Fix the grammatical error in this sentence

**Comment by charlesndirutu33** [comment]
@wilkistermumbi, I have fixed the issues and other grammatical errors.

Thanks.

**Comment by WanjaMIKE** [comment]
Merged #7318 into master

**Pull request merged**